test/config_disseminate_test.go

```diff
@@ -153,3 +169,196 @@ func TestConfigDisseminate(t *testing.T) {
 		require.True(t, protoutil.IsConfigBlock(lastBlock))
 	}
 }
+
+func TestConfigTXDisseminationWithVerification(t *testing.T) {
+	// Compile Arma
+	armaBinaryPath, err := gexec.BuildWithEnvironment("github.com/hyperledger/fabric-x-orderer/cmd/arma", []string{"GOPRIVATE=" + os.Getenv("GOPRIVATE")})
+	defer gexec.CleanupBuildArtifacts()
+	require.NoError(t, err)
+	require.NotNil(t, armaBinaryPath)
+
+	// Generate the configuration with clientSignatureVerificationRequired = True
+	dir := t.TempDir()
+	configPath := filepath.Join(dir, "config.yaml")
+	numOfParties := 4
+	numOfShards := 2
+	orgNum := 1
+	netInfo := testutil.CreateNetwork(t, configPath, numOfParties, numOfShards, "mTLS", "mTLS")
+	require.NoError(t, err)
+	numOfArmaNodes := len(netInfo)
+
+	armageddon.NewCLI().Run([]string{"generate", "--config", configPath, "--output", dir, "--clientSignatureVerificationRequired"})
+
+	// Run Arma nodes
+	// NOTE: if one of the nodes is not started within 10 seconds, there is no point in continuing the test, so fail it
+	readyChan := make(chan struct{}, numOfArmaNodes)
+	armaNetwork := testutil.RunArmaNodes(t, dir, armaBinaryPath, readyChan, netInfo)
+	testutil.WaitReady(t, readyChan, numOfArmaNodes, 10)
+
+	// Create a broadcast client
+	uc, err := testutil.GetUserConfig(dir, 1)
+	assert.NoError(t, err)
+	assert.NotNil(t, uc)
+
+	broadcastClient := client.NewBroadcastTxClient(uc, 10*time.Second)
+	defer broadcastClient.Stop()
+
+	// Prepare a Config TX, i.e. an envelope signed by an admin of org1
+	// the envelope.Payload contains marshaled bytes of configUpdateEnvelope, which is an envelope with Header.Type = HeaderType_CONFIG_UPDATE, signed by majority of admins
+
+	// Create an admin signer serves as the client which signs the config transaction
+	keyPath := filepath.Join(dir, "crypto", "ordererOrganizations", fmt.Sprintf("org%d", orgNum), "users", "admin", "msp", "keystore", "priv_sk")
+	signer, err := CreateSigner(keyPath)
+	require.NoError(t, err)
+	require.NotNil(t, signer)
+
+	certPath := filepath.Join(dir, "crypto", "ordererOrganizations", fmt.Sprintf("org%d", orgNum), "users", "admin", "msp", "signcerts", fmt.Sprintf("Admin@Org%d-cert.pem", orgNum))
+	certBytes, err := os.ReadFile(certPath)
+	require.NoError(t, err)
+	require.NotNil(t, certBytes)
```

Comment:
there is a bit of a duplication here, as all this is also done within `createConfigTX`. instead you could have just just indicated to  `createConfigTX` which admin id submits, and thus signs the TX itself, no?